sa/database.go

```diff
@@ -12,11 +13,33 @@ import (
 	blog "github.com/letsencrypt/boulder/log"
 )
 
+// DbSettings contains tuning settings for the database/sql driver. The
+// zero value of each field means use the default setting from
+// database/sql. ConnMaxIdleTime and ConnMaxLifetime should be set lower
+// than their mariab counterparts interactive_timeout and wait_timeout.
+type DbSettings struct {
+	// MaxOpenConns sets the maximum number of open connections to the
+	// database. If left unspecified connections to the database default
+	// to unlimited.
+	MaxOpenConns int
+	// MaxIdleConns is optional. When n > 0 it sets the size of the idle
+	// connection pool. The value is a pointer to an int because if n <=
+	// 0, no idle connections are retained. If nil is specified the sql
+	// driver will use the package default instead.
+	MaxIdleConns *int
+	// ConnMaxLifetime is optional. When n > 0 it sets the maximum
```

Comment:
Because this is documented as "optional", we should ensure _here in our code_ that we don't actually call through to `db.SetConnMaxLifetime` unless this has a non-default value.

Right now, the four `var setFoo` functions unconditionally call their corresponding db methods, and the `NewDbMapFromConfig` function similarly unconditionally calls all four of those. The golang `sql` package does not document explicit defaults for max idle time, max lifetime, and max open conns (it only documents a default of 2 for max idle conns). In case that package introduces new defaults in the future, we should ensure we're actually falling through to those defaults, instead of manually setting our own values (which happen to currently correspond with today's defaults).